packages/react-native-web/src/exports/Image/index.js

```diff
@@ -194,54 +157,36 @@ const Image: React.AbstractComponent<ImageProps, React.ElementRef<typeof View>>
     const selectedSource = shouldDisplaySource ? source : defaultSource;
     const displayImageUri = resolveAssetUri(selectedSource);
     const imageSizeStyle = resolveAssetDimensions(selectedSource);
-    const backgroundImage = displayImageUri ? `url("${displayImageUri}")` : null;
-    const backgroundSize = getBackgroundSize();
+    const crossOrigin = typeof selectedSource === 'object' ? selectedSource.crossOrigin : undefined;
 
     // Accessibility image allows users to trigger the browser's image context menu
-    const hiddenImage = displayImageUri
+    const image = displayImageUri
       ? createElement('img', {
-          alt: accessibilityLabel || '',
-          classList: [classes.accessibilityImage],
+          alt: accessibilityLabel !== null ? accessibilityLabel || '' : undefined,
+          classList: [classes.image],
+          crossOrigin: crossOrigin,
           draggable: draggable || false,
           ref: hiddenImageRef,
-          src: displayImageUri
+          src: displayImageUri,
+          style: [style, resizeModeStyles[resizeMode], { filter }]
         })
       : null;
 
-    function getBackgroundSize(): ?string {
-      if (hiddenImageRef.current != null && (resizeMode === 'center' || resizeMode === 'repeat')) {
-        const { naturalHeight, naturalWidth } = hiddenImageRef.current;
-        const { height, width } = layout;
-        if (naturalHeight && naturalWidth && height && width) {
-          const scaleFactor = Math.min(1, width / naturalWidth, height / naturalHeight);
-          const x = Math.ceil(scaleFactor * naturalWidth);
-          const y = Math.ceil(scaleFactor * naturalHeight);
-          return `${x}px ${y}px`;
-        }
-      }
-    }
-
-    function handleLayout(e) {
-      if (resizeMode === 'center' || resizeMode === 'repeat' || onLayout) {
-        const { layout } = e.nativeEvent;
-        onLayout && onLayout(e);
-        updateLayout(layout);
-      }
-    }
-
     // Image loading
+    // NOTE: in order to prevent costly reloads when objects aren't equal, we only check the resolved URI here
+    // however, what we really want is to deepEqual test the source var on all props that might affect loading.
```

Comment:
I've got the same problem for my PR: #2442 where it's possible to have `source.headers` 

The way I handle it is to extract `source` to local state, and only change it when the input source have deeply changed

https://github.com/necolas/react-native-web/blob/abbfe8f8eb81d859b274f87822b30f9bfa580031/packages/react-native-web/src/exports/Image/index.js#L369-L380

This will in turn trigger the loading effect hook

To make this work I've changed the `resolveAssetUri` to `resolveSource` and always work with the object shape of the source. This is to make the rest of the logic simpler and not having to guess whether the source is `number` or anything else 
But this results in some conflicts with your PR 
Can you take a look and give you opinion on how to move forward - maybe we can move the hook / change detection logic to your PR

Reply:
On my phone while traveling so not diving into the code too much. But in general, I think it would be great to patch on top of this. I strongly believe a broader image rewrite is merited, and this solves several tasks on that front.

The only issue is that I'm not sure if/when this will end up in master or the other image work will land, so it may take a bit of finagling and ordering to get all the image related patches and work together.